dali/operators/decoder/video/video_decoder_base.h

```diff
@@ -60,19 +61,59 @@ class DLL_PUBLIC VideoDecoderBase {
     return shape;
   }
 
+
   /**
-   * @brief Decode sample with index `idx` to `output` tensor.
+   * Decode given number of frames from a FramesDecoder.
+   *
+   * This function allows optional padding of the output data, when
+   * there's not enough frames inside FramesDecoder to fill the `num_frames` specified.
+   *
+   * @param output The SampleView in which the decoded sequence will be put.
+   * @param sample_idx Index of the encoded video that shall be decoded.
+   * @param num_frames How many frames shall be decoded.
+   * @param pad_value What to (optionally) pad the partial sequence with. If no value has been
+   *                  provided, padding will not happen.
+   * @return False, if less than `num_frames` have been decoded.
    */
-  void DecodeSample(SampleView<OutBackend> output, int64_t idx) {
-    auto &frames_decoder = *frames_decoders_[idx];
-    int64_t num_frames = output.shape()[0];
+  bool DecodeFrames(SampleView<OutBackend> output, int64_t sample_idx, int64_t num_frames,
+                    std::optional<SampleView<OutBackend>> pad_value = std::nullopt,
+                    cudaStream_t stream = 0) {
+    auto &frames_decoder = *frames_decoders_[sample_idx];
     int64_t frame_size = frames_decoder.FrameSize();
+
+    DALI_ENFORCE(!pad_value.has_value() || pad_value->shape().num_elements() == frame_size,
+                 make_string("Provided pad_value has improper number of elements. Expected: ",
+                             frame_size, "; Actual: ", pad_value->shape().num_elements()));
+
     uint8_t *output_data = output.template mutable_data<uint8_t>();
-    for (int f = 0; f < num_frames; ++f) {
+
+    int64_t f = 0;
+    // Work until:
+    //    (a) There are no more frames, or
+    //    (b) Sufficient number of frames has been decoded.
+    for (; f < num_frames && frames_decoder.NextFrameIdx() != -1; f++) {
       frames_decoder.ReadNextFrame(output_data + f * frame_size);
     }
+    assert(f <= num_frames);
+    bool full_sequence_decoded = f == num_frames;
+    // If there's an insufficient number of frames, pad if requested.
+    for (; f < num_frames && pad_value.has_value(); f++) {
+      CUDA_CALL(cudaMemcpy(output_data + f * frame_size, pad_value->raw_data(), frame_size,
```

Comment:
Do you want `pad_value` to house the whole frame?

Reply:
That's correct. I'll put it in the documentation, since it's probably not clear.